lib/Backends/Interpreter/Interpreter.cpp

```diff
@@ -309,15 +309,18 @@ bool Interpreter::isOpSupported(const NodeInfo &NI) const {
     return NI.allInputsAndOutputsHaveSameElemKind({ElemKind::Int8QTy});
 
   case Kinded::Kind::ConvertToNodeKind:
-    return ((NI.getInElemTy(ConvertToNode::InputIdx) == ElemKind::FloatTy) &&
-            ((NI.getOutElemTy(ConvertToNode::ResultIdx) ==
-              ElemKind::Float16Ty) ||
-             (NI.getOutElemTy(ConvertToNode::ResultIdx) ==
-              ElemKind::Int64ITy))) ||
-           ((NI.getInElemTy(ConvertToNode::InputIdx) == ElemKind::Float16Ty) &&
-            (NI.getOutElemTy(ConvertToNode::ResultIdx) == ElemKind::FloatTy)) ||
-           ((NI.getInElemTy(ConvertToNode::InputIdx) == ElemKind::Int64ITy) &&
-            (NI.getOutElemTy(ConvertToNode::ResultIdx) == ElemKind::FloatTy));
+    return (((NI.getInElemTy(ConvertToNode::InputIdx) == ElemKind::FloatTy) ||
```

Comment:
I must admit this is a bit hard to read. Can you add a comment here explaining expectations, like what input types could be and what output types would be supported for those inputs.

Reply:
Actually I just realized that we also miss float16<-> conversions. The point here is to make all types `{int32, int64, float16, float32}` convertable to one another. I'll add support for float16, and this check will simplify a lot after that..